drivers/wireless/lpwan/rn2xx3/rn2xx3.c

```diff
@@ -0,0 +1,1508 @@
+{
+  ssize_t length = 0;
+  ssize_t i = 0;
+
+  for (; i < nbytes; )
+    {
+      length = file_read(&priv->uart, &buf[i], nbytes - i);
+
+      if (length < 0)
+        {
+          return length;
+        }
+
+      i += length;
+
+      if (buf[i] == '\n' || (buf[i - 1] == '\n' && i > 0))
+        {
+          break;
+        }
+    }
+
+  return i + 1; /* Number of bytes read */
+}
+
+/****************************************************************************
+ * Name: mac_pause
+ *
+ * Description:
+ *   Pauses the MAC layer. Required before every transmission/receive.
+ *
+ ****************************************************************************/
+
+static int mac_pause(FAR struct rn2xx3_dev_s *priv)
+{
+  ssize_t length = 0;
+  char response[30];
+
+  /* Issue pause command */
+
+  length =
+      file_write(&priv->uart, "mac pause\r\n", sizeof("mac pause\r\n") - 1);
+  if (length < 0)
+    {
+      return length;
+    }
+
+  /* Wait for response of watchdog timer timeout */
+
+  length = read_line(priv, response, sizeof(response));
+  if (length < 0)
+    {
+      return length;
+    }
+
+  response[length] = '\0'; /* Avoid strstr overrun */
+
+  /* Check for pause duration */
+
+  if (strstr(response, MAC_PAUSE_DUR) == NULL)
+    {
+      return -EIO;
+    }
+
+  return 0;
+};
+
+/****************************************************************************
+ * Name: get_command_err
+ *
+ * Description:
+ *   Parses the command error response from the radio module and converts it
+ *   into an error code. 0 for 'ok', -EINVAL for 'invalid_param' and -EBUSY
+ *   for 'busy'. If an unknown error occurs, -EIO is returned.
+ *
+ ****************************************************************************/
+
+static int get_command_err(FAR struct rn2xx3_dev_s *priv)
+{
+  char response[18];
+  ssize_t length;
+
+  length = read_line(priv, response, sizeof(response));
+  if (length < 0)
+    {
+      return length;
+    }
+
+  response[length] = '\0';
+
+  if (strstr(response, "ok"))
+    {
+      /* Do nothing, this is good */
+
+      return 0;
+    }
+  else if (strstr(response, "invalid_param"))
+    {
+      wlerr("RN2xx3 invalid_param\n");
+      return -EINVAL;
+    }
+  else if (strstr(response, "busy"))
+    {
+      wlerr("RN2xx3 busy");
+      return -EBUSY;
+    }
+
+  wlerr("Unknown error");
+  return -EIO;
+}
+
+/****************************************************************************
+ * Name: rn2903_txpwrlevel
+ *
+ * Description:
+ *    Get the transmission power level that is greater than or equal to the
+ *    requested transmission power in dBm. These levels are for the RN2903.
+ *    The true dBm level that was set is returned in the `dbm` pointer.
+ *
+ ****************************************************************************/
+
+static int8_t rn2903_txpwrlevel(FAR float *dbm)
+{
+  if (*dbm <= 3.0f)
+    {
+      *dbm = 3.0f;
+      return 2;
+    }
+  else if (3.0f < *dbm && *dbm <= 4.0f)
```

Comment:
just a note: you don't need else if you return in all if cases

Reply:
If I remove `else` the style checker wants a blank line between each closing brace, so this will save space.

Reply:
Ah yes, fair enough, code style in nuttx is indeed a bit "wide" ;)